modules/text/src/ocr_holistic.cpp

```diff
@@ -0,0 +1,1064 @@
+
+        Mat preprocessed;
+        // preprocesses each image in the inputImageList and push to preprocessedImList
+        for(size_t imgNum=0;imgNum<inputImageList.size();imgNum++)
+        {
+            this->preprocess(inputImageList[imgNum],preprocessed);
+            preProcessedImList.push_back(preprocessed);
+        }
+        // set input data blob in dnn::net
+        net_->setInput(blobFromImages(preProcessedImList,1, this->inputGeometry_), "data");
+
+        float*outputMatData=(float*)(outputMat.data);
+       //Mat outputNet(inputImageList.size(),this->outputSize_,CV_32FC1,outputMatData) ;
+       Mat outputNet = this->net_->forward();
+       outputNet = outputNet.reshape(1, 1);
+
+       float*outputNetData=(float*)(outputNet.data);
+
+       memcpy(outputMatData,outputNetData,sizeof(float)*this->outputSize_*inputImageList.size());
+
+#endif
+    }
+
+#ifdef HAVE_DNN
+    Ptr<Net> net_;
+#endif
+    // hard coding input image size. anything in DNN library to get that from prototxt??
+   // Size inputGeometry_;//=Size(100,32);
+    int minibatchSz_;//The existence of the assignment operator mandates this to be nonconst
+    int outputSize_;
+    //Size outputGeometry_;//= Size(1,1);
+    //int channelCount_;
+   // int inputChannel_ ;//=1;
+    const int _inputHeight =32;
+    const int _inputWidth =100;
+    const int _inputChannel =1;
+public:
+    DeepCNNOpenCvDNNImpl(const DeepCNNOpenCvDNNImpl& dn):
+        minibatchSz_(dn.minibatchSz_),outputSize_(dn.outputSize_){
+        channelCount_=dn.channelCount_;
+        inputGeometry_=dn.inputGeometry_;
+        //Implemented to supress Visual Studio warning "assignment operator could not be generated"
+#ifdef HAVE_DNN
+        this->net_=dn.net_;
+#endif
+    }
+    DeepCNNOpenCvDNNImpl& operator=(const DeepCNNOpenCvDNNImpl &dn)
+    {
+#ifdef HAVE_DNN
+        this->net_=dn.net_;
+#endif
+        this->setPreprocessor(dn.preprocessor_);
+        this->inputGeometry_=dn.inputGeometry_;
+        this->channelCount_=dn.channelCount_;
+        this->minibatchSz_=dn.minibatchSz_;
+        this->outputSize_=dn.outputSize_;
+        this->preprocessor_=dn.preprocessor_;
+        this->outputGeometry_=dn.outputGeometry_;
+        return *this;
+        //Implemented to supress Visual Studio warning "assignment operator could not be generated"
+    }
+
+    DeepCNNOpenCvDNNImpl(String modelArchFilename, String modelWeightsFilename,Ptr<ImagePreprocessor> preprocessor, int maxMinibatchSz)
+        :minibatchSz_(maxMinibatchSz)
+    {
+
+        CV_Assert(this->minibatchSz_>0);
+        CV_Assert(fileExists(modelArchFilename));
+        CV_Assert(fileExists(modelWeightsFilename));
+        CV_Assert(!preprocessor.empty());
+        this->setPreprocessor(preprocessor);
+#ifdef HAVE_DNN
+
+        this->net_ = makePtr<Net>(readNetFromCaffe(modelArchFilename,modelWeightsFilename));
```

Comment:
I think this is the most **important** part of the whole pull-request! You should mention it right at the top of the class and maybe link to this page - http://docs.opencv.org/3.3.0/d6/d87/group__dnnLayerList.html saying this will work as long as the layer list includes your network's layers. So, did you have to add any new layers in core DNN to get the text network to work? These assumptions should go in the class level comments.

Reply:
I will update the comment section